app/src/main/java/com/zhuinden/jetpacknavigationdaggersavedstatehandleftueexperiment/utils/ViewModelUtils.kt

```diff
@@ -13,68 +13,44 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+@file:Suppress("UNCHECKED_CAST")
+
 package com.zhuinden.jetpacknavigationdaggersavedstatehandleftueexperiment.utils
 
-import android.os.Bundle
 import androidx.annotation.IdRes
 import androidx.fragment.app.Fragment
-import androidx.fragment.app.createViewModelLazy
+import androidx.fragment.app.viewModels
 import androidx.lifecycle.AbstractSavedStateViewModelFactory
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
-import androidx.navigation.fragment.findNavController
-import androidx.savedstate.SavedStateRegistryOwner
-
-inline fun <reified T : ViewModel> SavedStateRegistryOwner.createAbstractSavedStateViewModelFactory(
-    arguments: Bundle,
-    crossinline creator: (SavedStateHandle) -> T
-): ViewModelProvider.Factory {
-    return object : AbstractSavedStateViewModelFactory(this, arguments) {
-        @Suppress("UNCHECKED_CAST")
-        override fun <T : ViewModel?> create(
-            key: String, modelClass: Class<T>, handle: SavedStateHandle
-        ): T = creator(handle) as T
-    }
-}
+import androidx.navigation.navGraphViewModels
 
 inline fun <reified T : ViewModel> Fragment.navGraphSavedStateViewModels(
     @IdRes navGraphId: Int,
-    crossinline creator: (SavedStateHandle) -> T
-): Lazy<T> {
-    // Wrapped in lazy to not search the NavController each time we want the backStackEntry
-    val backStackEntry by lazy { findNavController().getBackStackEntry(navGraphId) }
-
-    return createViewModelLazy(T::class, storeProducer = {
-        backStackEntry.viewModelStore
-    }, factoryProducer = {
-        backStackEntry.createAbstractSavedStateViewModelFactory(
-            arguments = backStackEntry.arguments ?: Bundle(), creator = creator
```

Comment:
I just realized I might be messing with this specific behavior here. I'm not setting the default args as the arguments of the backstack entry, I'm setting it as the arguments of the fragment where the `ViewModel` is being created. I guess this might be a problem... 🤔

Reply:
If this is indeed a problem (I'm honestly not sure yet), we could reimplement `navGraphSavedStateViewModels()` like this:

```kotlin
inline fun <reified T : ViewModel> Fragment. navGraphSavedStateViewModels(
  @IdRes navGraphId: Int,
  crossinline viewModelProducer: (SavedStateHandle) -> T
): Lazy<T> {
  val backStackEntry by lazy { findNavController().getBackStackEntry(navGraphId) }
  return createViewModelLazy(T::class,
    storeProducer = { backStackEntry.viewModelStore },
    factoryProducer = {
      object : AbstractSavedStateViewModelFactory(this, backStackEntry.arguments) {
        override fun <T : ViewModel> create(key: String, modelClass: Class<T>, handle: SavedStateHandle) =
          viewModelProducer(handle) as T
      }
    }
  )
}
```

Reply:
It is indeed meant to be the arguments that potentially belong to the graph created by an action, isn't the alternative the same as what I have now?

Reply:
Yeah, pretty much :(

My current implementation for this would be useful if we just wanted a nav graph scoped `ViewModel` without the  `SavedStateHandle`. I still like how the other extensions got simpler, though, so I still see value on this PR. Nav graph scope + saved state is the only case where things get more complicated because of the `defaultArgs`, so I think it's nice having all the other extensions extremely simple and straightforward and this one being the only one more complicated.

Reply:
Seems like this should be fine https://twitter.com/ianhlake/status/1261308355085123585?s=20